docs/cluster.txt

```diff
@@ -1,5 +1,158 @@
-.. _cluster:
+.. _cluster-browser:
 
 ===============
 Cluster Browser
 ===============
+
+The CrateDB admin UI comes with a *cluster browser* that allows you to inspect
+all of the nodes that are in your `cluster`_.
+
+.. rubric:: Table of Contents
+
+.. contents::
+   :local:
+
+Screenshot
+==========
+
+Here's what a single node cluster looks like:
+
+.. image:: cluster-simple.png
+
+In this example, there is one node named *Monte Civetta*. When selected from
+the left-hand sub-navigation menu, information about a node is displayed.
+
+Features
+========
+
+.. |nbsp| unicode:: 0xA0
+   :trim:
+
+The cluster browser features:
+
+:Node |nbsp| listing:
+
+  The list of nodes gives basic summary information for each node: node name,
+  hostname, CrateDB version, and a visual icon indicating whether the node is a
+  ``master node`` or not and the node :ref:`data status <status-bar>` health
+  check.
+
+:Node |nbsp| list |nbsp| ordering:
+
+  The list of nodes can be ordered by name or node health.
+
+:Detailed node information:
+
+  Selecting a node from the node list will display details about that particular
+  node.
+
+The following subsections explain the detailed node information.
+
+Overview
+--------
+
+:Name:
+
+  The name of the node.
+
+  CrateDB automatically names unnamed nodes.
+
+:Hostname:
+
+  The hostname of the node.
+
+:CrateDB Version:
+
+  The version of CrateDB the node is running.
+
+  This is important information when performing a `rolling upgrade`_.
+
+:REST URL:
+
+  The URL of the node's `HTTP endpoint`_.
+
+:CPU Usage:
+
+  A visual indicator of system CPU utilization.
+
+:Heap Usage:
+
+  A visual indicator of allocated *Java Virtual Machine* (JVM) `heap`_
+  utilization.
+
+:Disk Usage:
+
+  A visual indicator of system disk space utilization.
+
+:CrateDB CPU Usage:
+
+  A visual indicator of allocated CrateDB CPU utilization.
+
+  TODO: is this correct? needs more explaining I think
+
+Load
+----
+
+:1min:
+
+  Average `load`_ over one minute.
+
+:5min:
+
+  Average load over five minutes.
+
+:15min:
+
+  Average load over fifteen minutes.
+
+CPU Cores
+---------
+
+:Cores:
+
+  The number of CPU cores.
+
+Shards
+------
+
+:Initializing:
+
+  The number of :ref:`shards <shards-browser>` currently being initialized.
+
+:Started:
+
+  The number of started (i.e. available) shards.
+
+:Reallocating:
+
+  The number of shards that are being moved to another node.
+
+:Post Recovery:
+
+  FIXME: what is this?
```

Comment:
@mxm can you help?

Reply:
Shards which data have been successfully recovered from disk (e.g. after starting a new node).

Reply:
This is from the `sys.shards` table.

Reply:
I'm not sure I follow

if I stop a node and then start it again, are all the shards on that node marked as "post-recovery"? (and if so, for what duration of time?)

or is there some subtlety I'm missing?

Reply:
More context:
https://www.elastic.co/guide/en/elasticsearch/reference/1.4/cat-recovery.html

>A recovery event occurs anytime an index shard moves to a different node in the cluster. This can happen during a snapshot recovery, a change in replication level, node failure, or on node startup. This last type is called a local gateway recovery and is the normal way for shards to be loaded from disk when a node starts up.

Reply:
thank you!

Reply:
`POST_RECOVERY` is the state of the shard when it has been recovered but is not yet `STARTED`. `STARTED` means it is accessible in the cluster through the cluster routing. It is the desired state of the shard because only then you can read/write data to it.